hashicorp/consul
# xds: don't attempt to load-balance sessions for local proxies

### Description
Previously, we'd begin a session with the xDS concurrency limiter regardless of whether the proxy was registered in the catalog or in the server's local agent state.

This caused problems for users who run `consul connect envoy` directly against a server rather than a client agent, as the server's locally registered proxies wouldn't be included in the limiter's capacity.

Now, the `ConfigSource` is responsible for beginning the session and we only do so for services in the catalog (bypassing the limiter for locally-registered proxies).

Fixes: https://github.com/hashicorp/consul/issues/15753

Also upgrades mockery to v2.15.0, as this was required to regenerate the mocks on Go 1.19.

### Testing & Reproduction steps
* Run a Consul cluster with >1 server.
* Register the majority of the proxies to 1 server.
* Run `consul connect envoy` for each proxy.
* Previously you'd see "this server has too many xDS streams open" in the Envoy logs.
* Now it should work as expected.
